std-bits/table/src/main/java/org/enso/table/data/column/storage/type/StorageType.java

```diff
@@ -1,7 +1,59 @@
 package org.enso.table.data.column.storage.type;
 
+import org.enso.base.polyglot.NumericConverter;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+
 /**
  * Represents an underlying internal storage type that can be mapped to the Value Type that is exposed to users.
  */
 public sealed interface StorageType permits AnyObjectType, BooleanType, DateType, DateTimeType, FloatType, IntegerType, TextType, TimeOfDayType {
+  /**
+   * @return the a common StorageType the two types.
+   */
+  static StorageType findGeneralType(StorageType firstType, StorageType secondType) {
+    if (firstType == null || firstType.equals(secondType)) {
+      return secondType;
+    }
+
+    return switch (firstType) {
+      case IntegerType i -> switch (secondType) {
+        case IntegerType i2 -> i.bits().toInteger() >= i2.bits().toInteger() ? i : i2;
+        case FloatType f -> FloatType.FLOAT_64;
+        default -> AnyObjectType.INSTANCE;
+      };
+      case FloatType f -> switch (secondType) {
+        case IntegerType i -> FloatType.FLOAT_64;
+        case FloatType f2 -> f.bits().toInteger() >= f2.bits().toInteger() ? f : f2;
+        default -> AnyObjectType.INSTANCE;
+      };
+      default -> AnyObjectType.INSTANCE;
```

Comment:
We will probably need to update this part once we get fixed-length text types fully supported in #5159, right?